packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -111,12 +111,25 @@ bool ArbCore::machineIdle() {
 
 ArbCore::message_status_enum ArbCore::messagesStatus() {
     auto current_status = message_data_status.load();
-    if (!core_error && current_status != MESSAGES_READY) {
+    if (current_status == MESSAGES_SUCCESS) {
         message_data_status = MESSAGES_EMPTY;
     }
+
     return current_status;
 }
 
+std::string ArbCore::messagesClearError() {
+    if (message_data_status != ArbCore::MESSAGES_ERROR) {
+        return "";
+    }
+
+    message_data_status = MESSAGES_EMPTY;
+    auto str = message_data_error_string;
+    message_data_error_string.clear();
```

Comment:
We should clear `message_data_error_string` before setting `message_data_status` back to empty, to ensure that another error isn't set inbetween the status being set and the error being cleared.

Reply:
:+1: